macaroonbakery/bakery/_checker.py

```diff
@@ -152,8 +149,7 @@ def _init_once(self, ctx):
             # other operations if the conditions succeed for those.
             declared, err = self._check_conditions(ctx, LOGIN_OP, conditions)
             if err is not None:
-                self._init_errors.append('cannot authorize login macaroon: ' +
-                                         err)
+                self._init_errors.append('cannot authorize login macaroon: ' + err)
```

Comment:
This should also be a real exception, not a string. I guess you could just `self._init_errors.append(AuthInitError('cannot authorize login macaroon: ' + err))`
￼

Reply:
perhaps non-pythonically, the check_first_party_caveat method returns a string error, not an exception, so I think this is OK as is.